### PR TITLE
Feat/api calls

### PR DESCRIPTION
**Pull request by deadbelly**
### What Type of PR Is This?
- [x] New Feature 
- [ ] Bug Fix
- [ ] Testing
- [ ] Refactoring/Clean Up 

#### What does this branch do? 
This branch adds api calls to handle most functionality without dummy data. There's a lot going on with this branch so I'll just list it out: 

- When a user signs in they'll fetch a unique id from our server, if they don't have one it'll add them to the table and generate one. 
- When a user saves a tour for the first time from the form, it will create a new tour and plan. If there is already a tour and plan it will update it. 
- When a user visits `/current-tours` they will see a list of tours they created that aren't marked complete. They can click on one of these tours to see it's corresponding plan and edit that form. 

Some notes:

- Patch function calls may be broken: I've removed the proxy server urls. This is because I think the issue might be that we're not sending the request from `https`. We can test this in our vercel app more easily than from localhost. 
- We still cannot delete any information. 
- Some form functionality still needs to be built (for example, the location and date are static because of a redesign in how form data was stored).
- We definitely need to clean up our file structure. I think a multiple files for our API calls (i.e. `plan_calls.js`, `user_calls.js`) and limiting util to helpers and cleaners would be better.
- I HAVE SKIPPED ALL TESTS: we should set up our PWA code, then go into our tests and stub all our API calls. This is at least one branch if not two's worth of work so I've simply disabled them for now. 

#### Have you tested your branch? How?
Just user-testing. We will need to write cypress tests soon.

#### Do all existing unit and integration tests pass? 
N/A 

#### Does this branch require special installation steps? 
I honestly don't think so but this branch is so big I think you should run `npm i` when you pull.

### CODE REVIEW

**Review by rachelbuchta — APPROVED** [review]
This was a huge doozy of a branch! Definitely need to go back and refactor some api calls but huge win getting all of this hooked up!

File: src/util.js [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+      "Content-Type": "application/json",
+      "Accept": "application/json",
+      "Authorization": `Bearer ${auth}`
+    },
+    body: JSON.stringify({})
+  }).then(response => checkResponse(response))
+}
+
+export const updateTour = (auth, userId, tourId, data) => {
+  return fetch(`https://cors-anywhere.herokuapp.com/https://backcountry-restapi.herokuapp.com/api/private/v1/user/${userId}/tour/${tourId}`,
```
Comment (rachelbuchta) [inline src/util.js:177]: Nice work around for this cors patch issue!

File: src/util.js [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+      }
+    }
+  ).then(response => checkResponse(response))
+}
+
+export const handleLogin = (auth, userData) => {
+  return getUser(auth, userData)
+    .catch(err => {
+      if (err.status === 404) {
+        console.log('hi')
```
Comment (tashiad) [inline src/util.js:93]: let's not forget to remove this later

**Review by tashiad — APPROVED** [review]
Nice work getting this all hooked up! I know there was a lot of troubleshooting involved. It's cool to see the solutions you came across.

**Pull request merged**